### DISCUSSION TITLE
1.6.0

### DISCUSSION THREAD

**Discussion by ilyhalight** [ilyhalight/voice-over-translation#728] [Обновления]
> [!WARNING]
> Версия не была протестирована в полном объеме, о любых известных проблемах пишите в [Issues](https://github.com/ilyhalight/voice-over-translation/issues), если их решение еще не было описано в [FAQ](https://github.com/ilyhalight/voice-over-translation/wiki/%5BRU%5D-FAQ).

## Список изменений

- Функционал запросов к API переписан с использованием vot.js
- Добавлена поддержка Kick (записи трансляций до 4 часов (/video/VIDEO_ID) + клипы)
- Добавлена поддержка Kodik
- Добавлена поддержка Reddit
- Добавлена поддержка Patreon
- Добавлена поддержка Archive.org (#679)
- Добавлена поддержка курсов без субтитров для Udemy
- Добавлена поддержка просмотра видео на стене в VK
- Добавлена поддержка VK Clips
- Добавлена поддержка YouTube Live (https://youtube.com/live/VIDEO_ID) (#657)
- Добавлена поддержка Bitchute embed
- Добавлена поддержка Apple Developer
- Добавлена поддержка русского домена XVideos (`xv-ru.com`)
- Для работы с Udemy больше не нужно вводить Udemy Access Token
- Обновлен список поддерживаемых сайтов для Invidious, ProxiTok, Peertube
- Убрана кнопка с прямых ссылок на видео с localhost / 127.0.0.1
- Добавлена возможность увеличения громкости аудио до 900% (#449)
- Улучшена скорость получения айди видео (#686)
- Исправлен жирный шрифт селекторов меню на некоторых сайтах
- Стандартный домен для проксирования изменен на `vot-worker.toil.cc`
- Теперь, при включенной опции "Не переводить с родного языка" проверяется, только, соответствие языка с которого переводится видео (ранее срабатывало, только, при одинаковой паре языков, например русский-русский, английкий-англиский и т.п.)
- Добавлена возможность выделения отдельных слов в субтитрах (#666)
- Добавлена возможность изменить размер шрифта субтитров (#720)
- Добавлена возможность изменить прозрачность фона субтитров (#720)
- Обновлена логика получения айди для clips.twitch.tv. Добавлена поддержка не только встроенных (embed) клипов
- Исправлено неправильное формирование параметра запроса для weverse
- Для dailymotion и yadisk итоговые ссылки заменены на короткие варианты
- Исправлен дополнительный селектор для twitter
- Работа с UI частично переписана на lit для работы с "Trusted Types"
- Исправлено накладывание друг на друга названия и плейсхолдера поля ввода
- Исправлен селектор для Bitchute
- Исправлен селектор для Facebook
- Proxytok переименован в Proxitok
- Стандартный формат загружаемых субтитров изменен на srt (#644)
- Фикс формирования строки с оставшимся временем перевода (#643)
- Добавлена индикация долгих операций (ожидание перевода, ожидание перевода текста, ожидание проверки аудиодорожки)
- Исправлен баг из-за которого реклама считалась за отдельные видео (#642)
- Фикс отображения кнопки для youku (#636)
- Исправлены некоторые визуальные недочеты
- Другие мелкие фиксы

## What's Changed
* Фикс кнопки на youku by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/636
* Попытка исправить #638 by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/640
* Revert "Попытка исправить #638" by @ilyhalight in https://github.com/ilyhalight/voice-over-translation/pull/641
* Исправление ошибок для #640 by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/642
* Форматирование субтитров в SRT by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/644
* Доработка VideoObserver и прочее  by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/650
* Проверка видимость видео для VideoObserver by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/651
* Небольшие правки by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/654
* Fix #657 by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/658
* Перенос обработчика mousemove и mouseup  на this.container by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/668
* Улучшение getUUID by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/674
* Доработка #674 by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/675
* Рефракторинг getSignature и небольшие изменения by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/676
* RegExp.exec() > String.match() by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/686
* упрощение regex by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/688
* Доработка логики обновления субтитров при переводе видео by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/690
* Доработка №1 by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/707
* Асинхронная обработка перевода и получения субтитров by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/694
* Доработка №2 by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/708
* Доработка updateTranslation by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/709
* Улучшение SubtitlesWidget by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/710
* Votjs integration by @ilyhalight in https://github.com/ilyhalight/voice-over-translation/pull/727

**Full Changelog**: https://github.com/ilyhalight/voice-over-translation/compare/1.5.3.1...1.6.0

<hr /><em>This discussion was created from the release <a href='https://github.com/ilyhalight/voice-over-translation/releases/tag/1.6.0'>1.6.0</a>.</em>

**Comment by FiLinX** [comment]
ха ребята, как вы вовремя то! )
я вот сейчас сидел и просматривал новинки со своей ленты подписок в ютубе, ну и как обычно я просто открываю в фоне вкладки с заинтересовавшими меня видео с той ленты.
короче понаоткрывал с чуть больше десятка, и начал просмотр с конца вкладок, в среднем чуть больше половины на иностранных языках - вобщем сижу смотрю и последовательно перехожу к предыдущим, и так вот досмотрел до трёх оставшихся, кликаю плей на третьей - буржуйская, слышу английский, кликаю в панельку VOTа - она сразу привычно встала в режим "отменить" без ожиданий  - но! пропал весь звук!
хм.. переоткрываю вкладку по F5 - а панельки VOTа уже и не вижу)
ладно, перехожу на вторую непросмотренную вкладку, тоже буржуйская - но панельки и там нету!
смотрю на первую вкладку - нема панельки.. 
чтож, думаю чего то лагануло в браузере - закрываю его, чищу кеши, запускаю брауз вновь, открываю скопированную ссылку от той третьей вкладки - и нету панельки VOTа!  открываю ещё несколько буржуйских ютубин из ленты похожих  -  но везде уже нету панельки!
ютуб прям в онлайне что-то для меня изменил..
хех.. вспоминаю тогда тутошнюю недавнюю иссю про аналогичную пропажу панельки в ютубе у людей, ну и только вот думаю пойду поставлю тоже данную там бету, а гляжу в релизах уже есть 1.6 - ставлю, проверяю - панелька в наличии и работает!
короче Спасибо что Вы есть ребята!!!

**Comment by FiLinX** [comment]
небольшая дописочка  -  в менюшке появился один не переведённый пункт - https://i.imgur.com/TRQOvAZ.jpeg

**Comment by FiLinX** [comment]
хех.. это изменение в ютубе - поломало и работу расширения Enhancer for YouTube™ ))
правда у меня стоит 2.0.121 его версия (и уже вот неработает, хотя полчаса назад всё живо было), но на сайте там 2.0.126 версия крайняя, попробую её, если даст установить mv3, а он требует не ниже 120 хрома, но у меня 118 ))
походу попал я..

**Comment by Vladik770S** [comment]
Пожалуйста добавьте возможность загрузки видео с аудио готового перевода на YouTube

**Reply by ilyhalight** [reply]
> Пожалуйста добавьте возможность загрузки видео с аудио готового перевода на YouTube

Не будет. Пользуйтесь https://github.com/FOSWLY/vot-cli

**Comment by NullVerdict** [comment]
На www.dailymotion.com нет кнопки, он даже в headers.json не прописан

**Reply by ilyhalight** [reply]
> На [www.dailymotion.com](http://www.dailymotion.com) нет кнопки, он даже в headers.json не прописан

Прописан, они поменяли поддомен для плеера/добавили распределение на несколько доменов
